libsupport/src/RDGVersion.cpp

```diff
@@ -0,0 +1,182 @@
+#include "katana/RDGVersion.h"
+
+#include "katana/Logging.h"
+
+namespace katana {
+
+RDGVersion::RDGVersion(
+    const std::vector<uint64_t>& vers, const std::vector<std::string>& ids)
+    : numbers_(vers), branches_(ids) {}
+
+RDGVersion::RDGVersion(uint64_t num) { numbers_.back() = num; }
+
+RDGVersion::RDGVersion(const std::string& src) {
+  KATANA_LOG_DEBUG_ASSERT(src.size() == 20);
+  char dest[kRDGVersionIDLength + 1];
+  char* token;
+
+  strncpy(dest, src.c_str(), kRDGVersionIDLength);
+  dest[kRDGVersionIDLength] = '\0';
+  token = strtok(dest, "_");
+
+  if (token != NULL) {
+    numbers_.clear();
+    branches_.clear();
+    do {
+      uint64_t val = strtoul(token, nullptr, 10);
+      numbers_.emplace_back(val);
+      token = strtok(NULL, "_");
+      if (token != NULL) {
+        branches_.emplace_back(token);
+        token = strtok(NULL, "_");
+      } else {
+        branches_.emplace_back(".");
+        break;
+      }
+    } while (token != NULL);
+  }
```

Comment:
I am not sure if that is necessary because it leaves the constructor empty and the parsing function is not used anywhere else.

Reply:
Reasonable. The language you are parsing still needs to be documented. You can either document it here or even better link to a page in `docs` which documents the version language in one place, so it can be referenced from various parts of the docs (since this language will be relevant to C++, Python, and probably sys admins).

This is a bit complex though, feel free to push this to another PR.